text/0000-result-expect.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: `result_expect`
+- Start Date: 2015-05-13
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add an `expect` method to the Result type, bounded to `E: Debug`
```

Comment:
@quantheory thanks for the elaboration.

> If Debug output is impossible even for a developer to understand, that's a whole separate problem.

It's more that the `Debug` output is sometimes less helpful in the case of unwrapping `Result`s:
 * If this is a `Result<_, String>` we get a string with newlines rendered as `\n` instead of just text.
 * If this is an `io::Result`, we get `Os(some_number)` instead of the actual error message. 

So if `expect` is to provide some convenience over `unwrap`, a middle ground between just spitting out a line number of the failed assert and using proper error handling, utilizing `Display` feels like better balance to me.